### PR TITLE
feat(sprint-details.component.html): fixed UI

### PR DESCRIPTION
**Pull request by AvinashMahanthi**
### Functionality:
fix completed sprint button UI

### Solution:
added some changes and reversed the filter sprint button and complete sprint button.

### Risk level:
- [X] high 
- [ ] medium
- [ ] low

### How to test:
ng serve
go to the dashboard
image

image

image

### CODE REVIEW

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]
Still need to work in UI.
The gap between button should not be there.
And also fix this in mobile view

File: worktrolly-ui/src/app/body/board/board.component.ts [hunk 1]
```diff
@@ -97,19 +98,29 @@ export class BoardComponent implements OnInit {
 
   changeSprintName(data: {newSprintNumber: number}) {
     this.currentSprintNumber = data.newSprintNumber;
+    if (this.currentSprintName != "S" + this.currentSprintNumber) {
+      this.currentSprintEnable = true;
+    }
+    else {
+      this.currentSprintEnable = false;
+    }
     this.currentSprintName = "S" + this.currentSprintNumber;
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/body/board/board.component.ts:107]: this is not exactly correct, check again

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]
complete this asap

File: worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.html [hunk 2]
```diff
@@ -2,32 +2,27 @@
     <div class="card">
         <div class="card-body" style="font-weight: bold; background-color: #8120f0; color:white;">
             <div class="row  align-items-center">
-                <div class="col-md-2 col-3">
+                <div class="col-md-2 col-4">
                     <h5 class="pl-2" style="font-size: 18px;">Sprint No:</h5>
                 </div>
-                <div class="col-md-4 col-2">
+                <div class="col-md-3 col-1">
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.html:8]: col-md-7, col-6

File: worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.html [hunk 3]
```diff
@@ -2,32 +2,27 @@
...
+                    <span class="material-icons align-middle ml-auto mr-2" *ngIf="Status === 'Completed' ">check_circle</span>
                     <div class="dropdown">
                         <button class="btn dropdown-toggle" type="button" id="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Filter</button>
-                        <div class="dropdown-menu px-5 py-4" id="filterMenu" aria-labelledby="dropdownMenuButton">
+                        <div class="dropdown-menu dropdown-menu-right px-4 py-3" id="filterMenu" aria-labelledby="dropdownMenuButton">
                             <div class="row py-3">
-                                <input class="col-md-8 col-8" style="font-size: 11px;" type="number" id="filterSprintNumberInput" placeholder="enter sprint number" [(ngModel)]="filterSprintNumber">
-                                <div class="col-md-4 col-4 pl-2">
+                                <input class="col-6" style="font-size: 11px;" type="number" id="filterSprintNumberInput" [(ngModel)]="filterSprintNumber">
+                                <div class="col-6 pl-2">
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.html:19]: line 11:  col-md-7 col-6

and first put the filter then completed

and completed button changed to be a status button where two option should be there under progress and completed, if status is under progress it simply update in database else same now u did.

and also align the icon using vertical alignment style

remove ml auto from line 12

and increase the size of input box in filter

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]
still need to work on UI

**Comment by AvinashMahanthi** [comment]
pc view:
image

mobile view:
Screenshot (210)

**Review by shivamgoyal7 — APPROVED** [review]
Ship it

File: worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.ts [hunk 4]
```diff
@@ -34,14 +35,32 @@ export class SprintDetailsComponent implements OnInit {
...
+    this.enableLoader = true;
+    const callable = this.functions.httpsCallable('updateSprintStatus');
+
+    try {
+      const result = await callable({ CurrentSprintName: this.currentSprintName, SprintStatus: this.sprintStatus }).toPromise();
+      console.log(this.sprintStatus);
+      console.log("Successfully updated Status");
+    } catch (error) {
+    this.enableLoader = false;
+    console.error("Error", error);
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.ts:63]: Reduce the code by reusing the methods with different arguments. So underProgress and completed can use the same method.

File: worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.html [hunk 5]
```diff
@@ -2,31 +2,34 @@
...
-                <div class=" col-md-2 col-2 ">
-                    <button class="btn " id="button" type="button" style="font-size: 14px;" (click)="completeSprint()">Completed</button>
-                    <span class="material-icons align-middle" *ngIf="Status === 'Completed' ">
-                        check_circle
-                    </span>
+                    <div class="btn-group" type="button" *ngIf="Status !== 'Completed' ">
+                        <button type="button" class="btn btn-primary dropdown-toggle" id="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Status</button>
+                        <div class="dropdown-menu">
+                            <div class="dropdown-item" type="button" (click)="updateSprintStatus()">Under progress</div>
+                            <div class="dropdown-item" type="button" (click)="completeSprint()">Completed</div>
```
Comment (viv9k) [inline worktrolly-ui/src/app/body/board/sprint-details/sprint-details.component.html:29]: Use the same methods with different arguments.

**Review by viv9k — CHANGES_REQUESTED** [review]
Please complete the required changes asap.

**Comment by Shrishti1802** [comment]
Resolve merge conflicts

**Comment by viv9k** [comment]
no action taken on this since last 3 days

**Pull request closed without merging**